src/UXClient/Components/DateTimePicker/DateTimePicker.ts

```diff
@@ -128,43 +132,50 @@ class DateTimePicker extends ChartComponent{
         this.fromMillis = fromMillis;
         this.toMillis = toMillis;
         this.onSet = onSet;
-        this.onCancel = onCancel;
+        this.onCancel = onCancel;   
         this.targetElement = d3.select(this.renderTarget)
             .classed("tsi-dateTimePicker", true);
         this.targetElement.html('');
         super.themify(this.targetElement, this.chartOptions.theme);
 
         let group = this.targetElement.append('div')
-            .classed('tsi-dateTimeGroup', true);
+            .classed('tsi-dateTimeGroup', true)
+            .on('keydown', (e) => {
+                if (d3.event.keyCode <= 40 && d3.event.keyCode >= 37) { //arrow key
+                    d3.event.stopPropagation();
+                }
+                if (d3.event.keyCode === 27 && this.chartOptions.dTPIsModal) { //escape
+                    this.onCancel();
+                    this.onSaveOrCancel();
+                }
+            });
+
         this.quickTimesPanel = group.append('div')
             .classed('tsi-quickTimesPanel', true);
         this.buildQuickTimesPanel();
 
         this.dateTimeSelectionPanel = group.append('div')
-            .classed('tsi-dateTimeSelectionPanel', true)
+            .classed('tsi-dateTimeSelectionPanel', true);
 
         this.timeControls = this.dateTimeSelectionPanel.append("div").classed("tsi-timeControlsContainer", true);
         this.calendar = this.dateTimeSelectionPanel.append("div").classed("tsi-calendarPicker", true);
         this.createTimezonePicker();
         var saveButtonContainer = this.dateTimeSelectionPanel.append("div").classed("tsi-saveButtonContainer", true);
         var self = this;
 
-        var onSaveOrCancel = () => {
-            this.isSettingStartTime = true;
```

Comment:
it now looks like using "enter" on save doesnt close when its a button?

Reply:
as far as I can tell it still does

Reply:
not in DateTimeButtonRange from what i can tell, it only closes in the availabilityChart